datasets/test/test_colours.py

```diff
@@ -0,0 +1,18 @@
+import json
+from datasets.colours import ColourDataset
+
+
+def test_colours():
+    with open('datasets/test/test_cases_colours.json', 'r') as file:
+        test_cases = json.load(file)
+
+    dataset = ColourDataset()
+
+    for test_case in test_cases:
+        questions = test_case['questions']
+        responses = test_case['responses']
+        expected_answers = test_case['expected_answers']
+
+        score, max_score, response = dataset.evaluate_correct(questions, responses, expected_answers)
+
+        assert score == max_score, f"Test Case: {test_case}\nExpected: {max_score}\nResult: {score}\nResponse: {response}"
```

Comment:
Because `test_case` shares names with the method's parameters, you could call it like `evaluate_correct(**test_case)` (just saying, it's good as you have it, and even clearer)

Apart from that, I guess from looking at the code that all tests in the JSON file are positive, right?

Reply:
Noted, I'll address that. 
Correct, all tests pass. There is one set of edge cases that I think I missed so I will investigate when I do the revisions.